packages/api/src/external/carequality/command/cq-organization/parse-cq-organization.ts

```diff
@@ -18,29 +22,31 @@ const XCA_DR_STRING = "ITI-39";
 const XDR_STRING = "ITI-41";
 type ChannelUrl = typeof XCPD_STRING | typeof XCA_DQ_STRING | typeof XCA_DR_STRING;
 
-export async function parseCQOrganization(
-  org: Organization
-): Promise<CQDirectoryEntryData | undefined> {
+export async function parseCQOrganization(org: Organization): Promise<CQDirectoryEntryData> {
+  const { log } = out(`parseCQOrganization`);
+
   const id = org.identifier?.[0]?.value;
-  if (!id) return undefined;
+  if (!id) throw new MetriportError("Missing ID on CQ Org", undefined, { org: stringify(org) });
+
+  const active = org.active;
+  if (active == undefined) {
+    throw new MetriportError("Missing active on CQ Org", undefined, { org: stringify(org) });
+  }
 
-  const lastUpdatedAtCQ = org.meta?.lastUpdated;
-  if (!lastUpdatedAtCQ) return undefined;
+  if (!org.meta?.lastUpdated) log("Missing lastUpdated at CQ Org, using current timestamp");
+  const lastUpdatedAtCQ = org.meta?.lastUpdated ?? buildDayjs().toISOString();
```

Comment:
not 100% sure about this move... we can discuss

Reply:
I think its good